### ISSUE TITLE
Add an unused module make resnet work bad

### ISSUE THREAD

**Issue by SmallPigPeppa** [Lightning-AI/lightning#13855]
## 🐛 Bug

I have try to add an unused module(e.g., resnet18) to resnet and train the resnet by some representation learning method (e.g., SimCLR). Although this module is excluded from weight decay and set requires_grad = False. The accuracy is worse than the initial. 

### To Reproduce

    class ResNet(nn.Module):

    def __init__(self, block, layers, num_classes=1000, zero_init_residual=False,
                 groups=1, width_per_group=64, replace_stride_with_dilation=None,
                 norm_layer=None):
        super(ResNet, self).__init__()
        if norm_layer is None:
            norm_layer = nn.BatchNorm2d
        self._norm_layer = norm_layer

        self.inplanes = 64
        self.dilation = 1
        if replace_stride_with_dilation is None:
            # each element in the tuple indicates if we should replace
            # the 2x2 stride with a dilated convolution instead
            replace_stride_with_dilation = [False, False, False]
        if len(replace_stride_with_dilation) != 3:
            raise ValueError("replace_stride_with_dilation should be None "
                             "or a 3-element tuple, got {}".format(replace_stride_with_dilation))
        self.groups = groups
        self.base_width = width_per_group
        self.conv1 = nn.Conv2d(3, self.inplanes, kernel_size=7, stride=2, padding=3,
                               bias=False)
        self.bn1 = norm_layer(self.inplanes)
        self.relu = nn.ReLU(inplace=True)
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
        self.layer1 = self._make_layer(block, 64, layers[0])
        self.layer2 = self._make_layer(block, 128, layers[1], stride=2,
                                       dilate=replace_stride_with_dilation[0])
        self.layer3 = self._make_layer(block, 256, layers[2], stride=2,
                                       dilate=replace_stride_with_dilation[1])
        self.layer4 = self._make_layer(block, 512, layers[3], stride=2,
                                       dilate=replace_stride_with_dilation[2])
        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
        self.fc = nn.Linear(512 * block.expansion, num_classes)
        # modified here
        from torchvision.models import resnet18, resnet50
        self.test=resnet18()

### Expected behavior

A worse Accuracy.

### Environment

- PyTorch Lightning Version: 1.5.4
- PyTorch Version: 1.10.2
- Python version: 3.8
- OS:  Linux

**Comment by carmocca** [comment]
This question does not seem related to PyTorch Lightning. I think your best course of action is to ask in the [PyTorch forums](https://discuss.pytorch.org/). Thank you!

**Comment by SmallPigPeppa** [comment]
Sorry to not describe this problem clearly, but it happens when I trained resnet18 by pytorch-lightning

**Comment by carmocca** [comment]
PyTorch Lightning does not interact at all with hyperparameters like weight decay (unless you tell it to!)

Unless the exact same model works when trained in a pure PyTorch but not when trained with PyTorch Lightning, this wouldn't be considered a Lightning issue
